src/Network/CurlResult.php

```diff
@@ -236,16 +236,29 @@ public function getHeader(string $field = '')
 			return $this->header;
 		}
 
-		$lines = explode("\n", $this->header);
-		foreach ($lines as $line) {
-			$parts = explode(':', $line);
-			$headerfield = array_shift($parts);
-			if (strtolower(trim($field)) == strtolower(trim($headerfield))) {
-				return trim(implode(':', $parts));
-			}
+		$field = strtolower(trim($field));
+
+		$headers = self::getHeaderArray();
+
+		if (isset($headers[$field])) {
+			return $headers[$field];
 		}
+	}
+
+	/**
+	 * Check if a specified header exists
+	 *
+	 * @param string $field header field
+	 *
+	 * @return boolean "true" if header exists
+	 */
+	public function headerExists(string $field)
```

Comment:
Please rename to `hasHeader` to match the syntax of `getHeader`.

Reply:
I don't like the sound of "hasHeader", I must confess :-)

Reply:
It's naming convention thing. Normally I know the rule of thumb that method names are starting with a verb and than the Name of the Field/Entity/State.

Like `getName()` , `isSuccess()`, `hasHeader()`, `createSomthing()`. It's for the homogeneity in the overall codebase.

Reply:
What about ```inHeader``` instead?

Reply:
That's fine! :)

Reply:
It's ok, but if you try to use it in an english sentence, I found the `hasHeader()` more intuitiv.

Like "if `CurlResult` has (the) Header `$field`, get (the) Header of `$field` and set it as variable".
Now it's "if `$field` is in (the) header of  `CurlResult', ..."

Reply:
I'm with @nupplaphil , and you'll never have to pronounce `hadHeader` anyway.